llvm/lib/Target/ARM/MCTargetDesc/ARMELFStreamer.cpp

```diff
@@ -793,20 +793,23 @@ void ARMTargetELFStreamer::switchVendor(StringRef Vendor) {
 
 void ARMTargetELFStreamer::emitAttribute(unsigned Attribute, unsigned Value) {
   getStreamer().setAttributeItem(Attribute, Value,
-                                 /* OverwriteExisting= */ true);
+                                 /* OverwriteExisting= */ true,
+                                 getStreamer().Contents);
```

Comment:
I don't like this API change, requiring non-AArch64 targets to pass a member variable of the stream back into the streamer every time. What about adding a new function to emit an attribute in a specific sub-section?

Reply:
Emitting into a specific subsection is not complicated.
All that is required is 1. first, create the subsection and add it to the subsections vector (AArch64TargetELFStreamer::emitSubsection) 2. then, use AArch64TargetELFStreamer::emitAttribute which will emit it into the correct subsection. 

Providing the data structure as an argument to the function rather then using the 'this' pointer (= the data structure that is in the class) data structure is, arguably, the more 'correct' way to construct a function in principal. Therefore it did not seems as a concern to change it. It might seems as a preferable solution to adding more functions to the base class.

Let me know what you think, no objection to changing it if you still think it is the better approach.

Reply:
Clarification: 
The part about the mechanism m for emitting attributes refer to ELF.
Asm attributes need to be emitted in order since they are being printed directly.

Reply:
Now that you've added the concept of the "active subsection", is this still needed, or could this be changed back to always emit into the active subsection?